src/ca.c

```diff
@@ -298,10 +298,9 @@ int create_capmt(SCAPMT *ca, int listmgmt, uint8_t *capmt, int capmt_len,
     capmt[pos++] = 0; // PI LEN 2 bytes, set 0
     capmt[pos++] = 0;
 
-    pos += CAPMT_add_PMT(capmt + pos, capmt_len - pos, pmt, cmd_id, added_only);
+    pos += CAPMT_add_PMT(capmt + pos, capmt_len - pos, pmt, cmd_id);
```

Comment:
there are 2 scenarios in which this is called:
1) DDCI adapter and as you mentioned, the unsubscribed pids are not present in the PMT (for which case the code is correct)
2) Regular CI/CAM: In that case the unsubscribed pids are also present in the PMTs, so we need to remove them.

This code was added for case 2. For 1, this code is supposed to add all the pids present in the PMT generated by the DDCI.

Reply:
I forgot about the second case, I'll drop this commit then 👍